pkg/cloud-init/cloud-init.go

```diff
@@ -690,3 +690,9 @@ func GenerateLocalData(vmi *v1.VirtualMachineInstance, instanceType string, data
 	log.Log.V(2).Infof("generated nocloud iso file %s", iso)
 	return nil
 }
+
+func GenerateGuestAgentBlacklistUserDataScript(commands string) string {
```

Comment:
You moved it to production codebase but I do not see any production code using it.

Reply:
correct, any other tests package that suits it please ? 
just felt cloud-init suit it, but it is indeed production and not tests one

i can also remove this commit from this PR and we can extract it on a follow-up from utils

Reply:
I am not sure what is the subject it belongs to.

It seems to be used in 4 tests. If you manage to understand its usefulness in those tests, perhaps we will know where it fits best.

Reply:
it is useful in those tests, i just moved it from utils, i will just drop this commit, and it can be done on a follow-up PR, because the PR scope is other function, and not this one